pinot-broker/src/main/java/org/apache/pinot/broker/broker/AccessControlFactory.java

```diff
@@ -29,10 +31,13 @@
   public static final String ACCESS_CONTROL_CLASS_CONFIG = "class";
 
   public abstract void init(PinotConfiguration confguration);
+  public abstract void init(ZkHelixPropertyStore<ZNRecord> propertyStore);
+
 
   public abstract AccessControl create();
 
-  public static AccessControlFactory loadFactory(PinotConfiguration configuration) {
+  public static AccessControlFactory loadFactory(PinotConfiguration configuration,
+      ZkHelixPropertyStore<ZNRecord> propertyStore) {
```

Comment:
@apucher Thanks agian, To keep backward compatibility with existing auth implementations, I have reconstructed relevant classes in controller, broker, and server. Please check the new design again.

Reply:
Great. IMO you can change the signature of the static `loadFactory` to include both configuration and property store. Since it's static it's not really part of the interface and there's no need to duplicate the code.

Rather that using `instanceof` I'd suggest to introduce an `init(PinotConfiguration, ZkHelixPropertyStore)` that defaults to invoking the legacy `init(PinotConfiguration)` method. That way, you achieve backwards compatibility and your new code can override the expanded method and take advantage of ZK access

Reply:
@apucher Thanks for your suggestions. I have changed the signature of the static loadFactory to include both configuration and property store. 

Also I have reconstructed AccessControlFactory and relevant class about Controller/Broker/Server. And introduce an init(PinotConfiguration, ZkHelixPropertyStore) that defaults to invoking the legacy init(PinotConfiguration) method Rather that using instanceof. Looking forward your review again.

Reply:
see comments below. I'd go as far as marking both init methods non-abstract, and by default implement the new one to call the old one.